libr/debug/debug.c

```diff
@@ -628,55 +806,42 @@ R_API int r_debug_continue_kill(RDebug *dbg, int sig) {
 		return false;
 	}
 	if (dbg->h && dbg->h->cont) {
-		if (dbg->in_recoil) {
-			/* if we are recoiling, we should not set the breakpoint that
-			 * caused us to stop.
-			 */
-			dbg->in_recoil = false;
-			r_bp_restore_except (dbg->bp, true, dbg->reason.addr);
-		} else {
-			r_bp_restore (dbg->bp, true); // set sw breakpoints
-		}
+		/* handle the stage-2 of breakpoints */
+		if (!r_debug_recoil (dbg, R_DBG_RECOIL_CONTINUE))
+			return false;
+
+		/* tell the inferior to go! */
 		ret = dbg->h->cont (dbg, dbg->pid, dbg->tid, sig);
-		dbg->reason.signum = 0;
-		retwait = r_debug_wait (dbg);
+
+		//XXX(jjd): why? //dbg->reason.signum = 0;
+
+		reason = r_debug_wait (dbg);
+
 #if __WINDOWS__
-		if (retwait != R_DEBUG_REASON_DEAD) {
+		if (reason != R_DEBUG_REASON_DEAD) {
+			// XXX(jjd): returning a thread id?!
```

Comment:
Yep that case will be problematic

> On 21 Jun 2016, at 02:09, Joshua J. Drake <notifications@github.com> wrote:
> 
> In libr/debug/debug.c:
> 
> >  #if __WINDOWS__
> > -		if (retwait != R_DEBUG_REASON_DEAD) {
> > +		if (reason != R_DEBUG_REASON_DEAD) {
> > +			// XXX(jjd): returning a thread id?!
> I don't think it's possible, but what if TID is 0?
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
>